### ISSUE TITLE
'ansible.vars.unsafe_proxy.AnsibleUnsafeText object' has no attribute

### ISSUE THREAD

**Issue by thedumbtechguy** [ansible/ansible#24264]
I've seen other similar posts, but I they seem not to apply to me. The role in question worked fine in 2.2.

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
Loop

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### CONFIGURATION
```
[defaults]

host_key_checking = False
ansible_managed = DO NOT MODIFY by hand. This file is under control of Ansible on {host}.
vault_password_file = /var/lib/semaphore/.vpf
```

##### OS / ENVIRONMENT
N/A

##### SUMMARY
When trying to create a list of users, it throws the an error about an invalid value error.

##### STEPS TO REPRODUCE

I am running this task

This is the task I am trying to run.

```yaml
- name: create user 
  user:
    name: "{{ item.username }}"
    password: "{{ item.password|default(omit) }}"
    shell: /bin/bash
  become: true
  with_items: '{{ users }}'
  no_log: true
```

This is the `users` variable. I have truncated the vault values for brevity.
```yaml
users:
  - username: ptadmin
    password: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...
    use_sudo: true
    use_ssh: false
  
  - username: ansibleremote
    password: "{{ petra_ansibleremote_password }}"
    use_sudo: true
    use_ssh: true
    public_key: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...

  - username: semaphore
    password: "{{ petra_ansibleremote_password }}"
    use_sudo: true
    use_ssh: true

  - username: frosty
    password: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...
    use_sudo: true
    use_ssh: true
    public_key: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...

  - username: thebeardedone
    password: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...
    use_sudo: true
    use_ssh: true
    public_key: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...

  - username: senanufc
    password: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...
    use_sudo: true
    use_ssh: true
    public_key: !vault-encrypted |
      $ANSIBLE_VAULT;1.1;AES256
      ...
```

I tried a `debug` before it and the results are in this [gist](https://gist.github.com/thedumbtechguy/33a2de0c2b841f360e89ec29559a4790).

I also tried with below and the same thing happens
```yaml
- name: test loop
  debug:
    msg: "{{ item.username }}"
  with_items: "{{ users }}"
```
##### EXPECTED RESULTS
Users created 

##### ACTUAL RESULTS

```
fatal: [petra-hq-dev-master]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'ansible.vars.unsafe_proxy.AnsibleUnsafeText object' has no attribute 'username'\n\nThe error appears to have been in '/etc/ansible/roles/thedumbtechguy.manage-users/tasks/create_users.yml': line 6, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: create user\n  ^ here\n"}
	to retry, use: --limit @/var/lib/semaphore/repository_1/playbooks/setup_new_hosts.retry
```

**Comment by Malet** [comment]
Having the same issue with the user module + vaulted passwords on ansible 2.3.0.0

**Comment by thedumbtechguy** [comment]
I figured out a workaround.
It seems it doesn't play nice if the vaulted variable is in a list or a dict, not sure which.

So I set a temporary variable first and use that in the list. e.g.

```yaml

_temp_user_password: !vault-encrypted |
  $ANSIBLE_VAULT;1.1;AES256
  63323263376630646666613735646531383639613066363830313234303061653736306339616334
  6334316339383635353039353339393331613265373732340a613230323431306261613161326433
  65323664373734373531393537306563623330313534346635383837663566383165366662306366
  3131333966313137610a633232383365653132653630626363323831383935646332646332346431
  6461

users:
  - username: user
    password: "{{ _temp_user_password}}"

```

**Comment by alikins** [comment]
This looks like a duplicate of #23846 and #24175 (fixed via https://github.com/ansible/ansible/pull/24312 in devel and 572c07ddb394b5798dc11c0c0215ba729dcd1850 in stable-2.3). Fix should be in 2.3.1 release.

duplicate_of #24175
